### PR TITLE
[WIP] set hvt_guest_mprotect for "hvt: Enforce that guest executable code is not also writable (W^X) #303"

### PR DESCRIPTION
**Pull request by adamsteen**
Hi @mato and @hannesm 

just looking for feedback about the setup of hvt_guest_mprotect and elf_load.

### CODE REVIEW

**Comment by adamsteen** [comment]
This is not going to work, we need a reference to the hot struct

File: tenders/hvt/hvt_openbsd.c [hunk 1]
```diff
@@ -136,6 +136,38 @@ struct hvt *hvt_init(size_t mem_size)
...
+
+    vmep->vmep_vm_id = hvb->vcp_id;
+    // sgpa = vmr->vmr_gpa(0x0) + (addr - vmr->vmr_va)
+    vmep->vmep_sgpa = (vaddr_t)addr - (vaddr_t)hvt->mem;
+    vmep->vmep_size = len;
+    vmep->vmep_prot = prot;
+
+    if (ioctl(hvb->vmd_fd, VMM_IOC_MPROTECT_EPT, vmep) < 0) {
+        warn("mprotect ept vmm ioctl failed - exiting");
+        return -1;
```
Comment (cfcs) [inline tenders/hvt/hvt_openbsd.c]: this returns without `free(vmep);`

**Comment by adamsteen** [comment]
crickets

**Comment by mato** [comment]
> crickets

?

**Comment by adamsteen** [comment]
Sorry, i think that was a copy and paste from somewhere, i definitely didn't mean to close this pull request! I apologise if any offence was caused.

**Comment by adamsteen** [comment]
Change title to [WIP], the solo5 code is pretty close, but the OpenBSD code needs some more investigation.

**Comment by mato** [comment]
@adamsteen For the record, I think this is (bar some naming/parameter ordering) the right approach from the Solo5 tender code PoV. I will make a separate PR with that part of the code only, once I am done with the current round of manifest/multi-device/ELF refactoring. Then you can continue this with only the OpenBSD-specific support to enable the functionality.

File: tenders/hvt/hvt_openbsd.c [hunk 2]
```diff
@@ -136,6 +136,42 @@ struct hvt *hvt_init(size_t mem_size)
...
 
+int hvt_guest_mprotect(void *t, void *addr, size_t len, int prot)
+{
+    int ret;
+
+    if(mprotect(addr, len, prot) == -1)
+        return -1;
+
+    ret = 0;
+#if defined(VMM_IOC_MPROTECT_EPT)
```
Comment (cfcs) [inline tenders/hvt/hvt_openbsd.c]: Is it worth supporting cases where EPT is not available?

**Comment by mato** [comment]
Superseded by #447.

**Pull request closed without merging**